engine/runtime/src/main/java/org/enso/interpreter/node/callable/dispatch/InvokeFunctionNode.java

```diff
@@ -224,4 +249,54 @@ public void setId(UUID id) {
   public UUID getId() {
     return functionCallInstrumentationNode.getId();
   }
+
+  /** Returns true if the given function is in the same project as this node. */
+  private boolean isInSameProject(Function function) {
+    var thisProjName = getThisProjectName();
+    var funcProjName = getFunctionProjectName(function);
+    if (thisProjName != null && funcProjName != null) {
+      return thisProjName.equals(funcProjName);
+    } else {
+      return false;
+    }
+  }
+
+  private String getThisProjectName() {
+    if (getRootNode() instanceof EnsoRootNode thisRootNode) {
+      var modScope = thisRootNode.getModuleScope();
+      if (modScope != null) {
+        var pkg = modScope.getModule().getPackage();
+        if (pkg != null) {
+          return pkg.name();
+        }
+      }
+    }
+    return null;
+  }
+
+  private String getFunctionProjectName(Function function) {
+    var modScope = getModuleScopeForFunction(function);
+    if (modScope != null) {
+      var pkg = modScope.getModule().getPackage();
+      if (pkg != null) {
+        return pkg.name();
+      }
+    }
+    return null;
+  }
+
+  private ModuleScope getModuleScopeForFunction(Function function) {
```

Comment:
I have seen the body of this method somewhere... Shouldn't we add `Function.getDefiningScope()` method into `Function` itself and make sure it is consistently used from all places?

Reply:
The only one somewhat similar usage is in https://github.com/enso-org/enso/blob/f8a546e85f180d90dedf14cc70b00b0ec6345044/engine/runtime-instrument-common/src/main/java/org/enso/interpreter/service/ExecutionService.java#L740-L774. But ultimately, that usage is different. Let's keep this method private in this class.